DIPs/7NNN-WGB.md

```diff
@@ -0,0 +1,168 @@
+and the user need only supply the arguments that matter for his use case. This
+is much more flexible than the current method of putting all the defaults at
+the
+end of the parameter list, and defaulting one means all the rest get defaulted.
+2. For a longish list of parameters, when the user
+finds himself counting parameters to ensure they line up, then named parameters
+can be most helpful.
+3. A step towards making it possible to replace the brace struct initialization syntax with
+a function-call-like construct. I.e. the initialization of structs is unified with
+struct literals and struct constructors.
+
+## Prior Work
+
+* DIP 88
+* DIP 1019
+* DIP 1020
+
+Both DIP 1019 and DIP 1020 detail prior work in other languages, which will not be
+repeated here.
+
+
+## Description
+
+This based on the recognition that D already has
+named parameters for [struct initializers](https://dlang.org/spec/struct.html#static_struct_init).
+Closely related are [array initializers](https://dlang.org/spec/arrays.html#static-init-static)
+and [associative array literals](https://dlang.org/spec/hash-map.html#static_initialization).
+
+The addition to the syntax is replacing the definition of `ArgumentList` for function call
+arguments with:
+
+```
+ArgumentList:
+   NamedArgument
+   NamedArgument ,
+   NamedArgument , NamedArgumentList
+
+NamedArgument:
+   Identifier : Argument
+   Argument
+```
+
+Assignment of arguments to parameters would work the same way as
+[assignments to fields](https://dlang.org/spec/struct.html#static_struct_init).
+
+Matching of the `NamedArgument`s to a function declaration proceeds in lexical order.
+For each `NamedArgument`:
+
+1. If an `Identifier` is present, it matches to the `Parameter` with the corresponding
+`Identifier`. If it does match any named `Parameter`, then it is an error.
+2. If an `Identifier` is not present, the `Parameter` matched is the one following
+the previous matched `Parameter`, or the first `Parameter` if none have been matched yet.
+3. Matching a `Parameter` more than once is an error.
+4. After the `NamedArgumentList` is exhausted, any unmatched `Parameter`s receive the
+corresponding default value specified for that `Parameter`. If there is no default value,
+it is an error.
+5. If there are more `NamedArgument`s than `Parameter`s, the remainder match the trailing
+`...` of variadic parameter lists, and `Identifier`s are not allowed.
+
+
+The set of matched functions becomes the overload set, to which the usual overload resolution
+is applied to select the best match.
+
+I.e. function resolution is done by constructing an argument list separately
+for
+each function before testing it for matching. Of course, if the parameter
+name(s) don't match, the function doesn't match. If a parameter has no
+corresponding argument, and no default value, then the function doesn't match.
+
+```
+void snoopy(T t, int i, S s);     // A
+void snoopy(S s, int i = 0; T t); // B
```

Comment:
I guess this is a typo? Otherwise I don't understand the semicolon or why i can have a default value with t requiring one.

Reply:
fixed